eZ/Publish/Core/Persistence/Legacy/Content/Handler.php

```diff
@@ -587,20 +598,29 @@ public function listVersions($contentId, $status = null, $limit = -1)
      */
     public function copy($contentId, $versionNo = null)
     {
+        $contentInfo = $this->loadContentInfo($contentId);
+
         $currentVersionNo = isset($versionNo) ?
             $versionNo :
-            $this->loadContentInfo($contentId)->currentVersionNo;
+            $contentInfo->currentVersionNo;
 
         // Copy content in given version or current version
         $createStruct = $this->mapper->createCreateStructFromContent(
             $this->load($contentId, $currentVersionNo)
         );
+
+        if (!isset($versionNo) && $currentVersionNo > 1) {
+            $createStruct->initialLanguageId = $this
+                ->languageHandler
+                ->loadByLanguageCode($contentInfo->mainLanguageCode)
+                ->id;
+        }
```

Comment:
I agree with @alongosz suggestion.  As someone said: 

> There are two hard things in computer science: cache invalidation, naming things 

I'm stuck on a second part.  So any ideas how to name parameter/method which says "get language code from $contentInfo instead of $versionInfo" are welcome :wink: